rosbag2_storage/test/rosbag2_storage/test_metadata_serialization.cpp

```diff
@@ -93,3 +93,39 @@ TEST_F(MetadataFixture, test_writing_and_reading_yaml)
     Eq(expected_second_topic.topic_metadata.serialization_format));
   EXPECT_THAT(actual_second_topic.message_count, Eq(expected_second_topic.message_count));
 }
+
+TEST_F(MetadataFixture, metadata_reads_v3)
```

Comment:
It would help the reader to have more specific test cases description: "metadata_reads_v3_check_offered_qos_profiles_empty", ditto for next one.

A line or two of comments explaining that this is a new feature in v4, and we make sure that the new attribute is properly initialized to zero would also be super helpful.

Reply:
Done